src/main/java/org/tools4j/hoverraft/event/AppendRequestHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.tools4j.hoverraft.event;
+
+import org.tools4j.hoverraft.message.AppendRequest;
+import org.tools4j.hoverraft.server.ServerContext;
+import org.tools4j.hoverraft.state.*;
+
+import java.util.Objects;
+
+public class AppendRequestHandler {
+    private final PersistentState persistentState;
+    private final VolatileState volatileState;
```

Comment:
you don't need those member variables cause they are passed to the onAppendRequest method via ServerContext